packages/dropzone/src/react.tsx

```diff
@@ -0,0 +1,347 @@
+/**
+ * This is a forked version of the react-dropzone package.
+ * See original source here: https://github.com/react-dropzone/react-dropzone
+ * The original package is licensed under the MIT license.
+ */
+
+import type {
+  ChangeEvent,
+  DragEvent,
+  HTMLProps,
+  KeyboardEvent,
+  MouseEvent,
+} from "react";
+import { useCallback, useEffect, useMemo, useReducer, useRef } from "react";
+import { fromEvent } from "file-selector";
+
+import {
+  acceptPropAsAcceptAttr,
+  allFilesAccepted,
+  initialState,
+  isEnterOrSpace,
+  isEventWithFiles,
+  isFileAccepted,
+  isIeOrEdge,
+  isValidQuantity,
+  isValidSize,
+  noop,
+  reducer,
+} from "./core";
+import type { DropEvent, DropzoneOptionsBase } from "./types";
+
+export type * from "./types";
+
+/**
+ * A React hook that creates a drag 'n' drop area.
+ *
+ * ### Example
+ *
+ * ```tsx
+ * function MyDropzone() {
+ *   const { getRootProps, getInputProps } = useDropzone({
+ *     onDrop: acceptedFiles => {
+ *       // do something with the File objects, e.g. upload to some server
+ *     }
+ *   });
+ *
+ *   return (
+ *     <div {...getRootProps()}>
+ *       <input {...getInputProps()} />
+ *       <p>Drag and drop some files here, or click to select files</p>
+ *     </div>
+ *   )
+ * }
+ * ```
+ */
+export function useDropzone({
```

Comment:
this should be easily portable to solid, vue etc now and can use all the helpers from `core.ts`